KAFKA-14375: Remove use of "authorizer-properties" from EndToEndAuthorizerTest

pprovenzano opened the pull request:
This removes use of a deprecated feature and instead has all ACL calls going through the brokers. This work is preliminary work needed before I can make them run in KRAFT mode.

https://issues.apache.org/jira/browse/KAFKA-14375

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

pprovenzano committed:
Remove use of "authorizer-properties" from EndToEndAuthorizerTest

This removes use of a depricated feature and instead has all ACL
calls going through the brokers. This work is preliminary work
needed before I can make them run in KRAFT mode.

soarez commented on core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala:
```diff
@@ -170,6 +187,19 @@ abstract class IntegrationTestHarness extends KafkaServerTestHarness {
     admin
   }
 
+
```
Extra line

pprovenzano replied:
Fixed!

soarez commented on core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala:
```diff
@@ -96,96 +90,53 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
+  def AclPrefixedGroupRead = new AclBinding(prefixedGroupResource,
+    new AccessControlEntry(clientPrincipal.toString, "*", AclOperation.READ, AclPermissionType.ALLOW))
+
+
```
Extra line

pprovenzano replied:
Fixed

soarez commented on core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala:
```diff
@@ -96,96 +90,53 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
+    new AccessControlEntry(clientPrincipal.toString, "*", AclOperation.CREATE, AclPermissionType.ALLOW))
+  def AclTopicDescribe(tptopicresource : ResourcePattern = topicResource) = new AclBinding(tptopicresource,
+    new AccessControlEntry(clientPrincipal.toString, "*", AclOperation.DESCRIBE, AclPermissionType.ALLOW))
+  def AclTopicRead(tptopicresource : ResourcePattern = topicResource) = new AclBinding(tptopicresource,
```
s/tptopicresource/topicResource/g

pprovenzano replied:
Fixed

soarez commented on core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala:
```diff
@@ -292,24 +255,35 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
+    val superuserAdminClient = createSuperuserAdminClient()
+
+    superuserAdminClient.createAcls(List(AclTopicWrite(tptopicresource), AclTopicCreate(tptopicresource), AclTopicDescribe(tptopicresource)).asJava).values
+    superuserAdminClient.createAcls(List(AclTopicRead(tptopicresource)).asJava).values
```
s/tptopicresource/topicResource/g

pprovenzano replied:
Fixed

soarez commented on core/src/test/scala/integration/kafka/api/PlaintextEndToEndAuthorizationTest.scala:
```diff
@@ -76,6 +77,15 @@ class PlaintextEndToEndAuthorizationTest extends EndToEndAuthorizationTest {
   }
 
+  /*
+   * The principal used for all authenticated connections to listernerName is always clientPrincipal.
```
s/listernerName/listenerName

pprovenzano replied:
Fixed

pprovenzano committed:
Cleanup extra space and bad camelcase.

pprovenzano committed:
Final test that needs updating to work with removing "authorizer-properties"
from EndToEndAuthorizerTest. Delegation tests still talk directly to ZK
and Delegation Tokens have not been added to work under KRAFT so I'm holding
off changing these tests to talk to brokers instead of ZK for initial setup.

omkreddy commented on core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala:
```diff
@@ -200,11 +150,23 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
+
     super.setUp(testInfo)
+
+    val superuserAdminClient = createSuperuserAdminClient()
```
Do we need to create `AclClusterAction, AclAlter` acls? since kafkaPrincipal is super user, we dont need these acls.

pprovenzano replied:
I don't know that we do. I know that it works with it, but I haven't tried testing without it.

pprovenzano replied:
I tested as we discussed and in fact it is not necessary. I removed the code at setUp().

omkreddy commented on core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala:
```diff
@@ -64,15 +63,10 @@ import scala.jdk.CollectionConverters._
   * SaslTestHarness here directly because it extends QuorumTestHarness, and we
   * would end up with QuorumTestHarness twice.
   */
+@Timeout(60)
```
why do we need Timeout?

pprovenzano replied:
In my testing which included the added KRAFT support, some tests failed without increasing the default timeout. @cmccabe suggested I put the timeout on the class.

omkreddy commented:
looks like we also need to update `DelegationTokenEndToEndAuthorizationWithOwnerTest`

pprovenzano commented:
I see some failed tests that seem to be flaky tests. What do we do here?

pprovenzano committed:
It is not necessary to give brokers ClusterActions and ClusterAlterAcls
if the brokers are set as superusers.

omkreddy commented on core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala:
```diff
@@ -96,96 +90,52 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
-  this.serverConfig.setProperty(KafkaConfig.ZkEnableSecureAclsProp, "true")
-  this.serverConfig.setProperty(KafkaConfig.AuthorizerClassNameProp, authorizerClass.getName)
+
+  def AclClusterAction = new AclBinding(clusterResource,
```
Can we remove the ClusterActionAndClusterAlterAcls, AclClusterAction and AclAlter fields now?

pprovenzano replied:
Done

omkreddy commented on core/src/test/scala/integration/kafka/api/SslEndToEndAuthorizationTest.scala:
```diff
@@ -81,4 +81,13 @@ class SslEndToEndAuthorizationTest extends EndToEndAuthorizationTest {
+  // This is really ugly and I'd like a better way to do this.
+  override def superuserSecurityProps(certAlias: String): Properties = {
+    val props = TestUtils.securityConfigs(Mode.CLIENT, securityProtocol, trustStoreFile,
+      certAlias, "server", clientSaslProperties, tlsProtocol)
```
Can we use "super-user" CN name?

pprovenzano replied:
Done

omkreddy commented on core/src/test/scala/integration/kafka/api/SslEndToEndAuthorizationTest.scala:
```diff
@@ -37,8 +37,8 @@ object SslEndToEndAuthorizationTest {
       peerPrincipal match {
-        case Pattern(name, _) =>
-          val principal = if (name == "server") name else peerPrincipal
+        case Pattern(name, extra) =>
```
extra => "cn"

pprovenzano replied:
Done

pprovenzano committed:
Cleanup changes from review.

Remove unneeded ACL defines in EndToEndAuthorizationTest.scala
Define a superuserCn in SslEndToEndAuthorizationTest.scala

pprovenzano commented:
I have manually tested all the failed tests from the checks and they all pass fine locally.

omkreddy reviewed (approved):
@pprovenzano Thanks for the PR. LGTM
